python/cuml/dask/feature_extraction/text/tfidf_transformer.py

```diff
@@ -0,0 +1,208 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import dask
+from toolz import first
+import dask.array
+
+from cuml.common import with_cupy_rmm
+from cuml.dask.common.base import BaseEstimator
+from cuml.dask.common.base import DelayedTransformMixin
+from cuml.dask.common.utils import wait_and_raise_from_futures
+from cuml.dask.common.func import reduce
+from cuml.dask.common.input_utils import DistributedDataHandler
+
+from cuml.feature_extraction.text import TfidfTransformer as s_TfidfTransformer
+
+
+class TfidfTransformer(BaseEstimator, DelayedTransformMixin):
+
+    """
+    Distributed TF-IDF  transformer
+
+    Examples
+    --------
+
+        import cupy as cp
+        from sklearn.datasets import fetch_20newsgroups
+        from sklearn.feature_extraction.text import CountVectorizer
+        from dask_cuda import LocalCUDACluster
+        from dask.distributed import Client
+        from cuml.dask.common import to_sparse_dask_array
+        from cuml.dask.naive_bayes import MultinomialNB
+        import dask
+        from cuml.dask.feature_extraction.text import TfidfTransformer
+
+        # Create a local CUDA cluster
+        cluster = LocalCUDACluster()
+        client = Client(cluster)
+
+        # Load corpus
+        twenty_train = fetch_20newsgroups(subset='train',
+                                shuffle=True, random_state=42)
+        cv = CountVectorizer()
+        xformed = cv.fit_transform(twenty_train.data).astype(cp.float32)
+        X = to_sparse_dask_array(xformed, client)
+
+        y = dask.array.from_array(twenty_train.target, asarray=False,
+                            fancy=False).astype(cp.int32)
+
+
+        mutli_gpu_transformer = TfidfTransformer()
+        X_transormed = mutli_gpu_transformer.fit_transform(X)
+        X_transormed.compute_chunk_sizes()
+
+        model = MultinomialNB()
+        model.fit(X_transormed, y)
+        model.score(X_transormed, y)
+
+    Output:
+
+    .. code-block:: python
+
+        array(0.93264981)
+
+    """
+
+    def __init__(self, client=None, verbose=False, **kwargs):
+
+        """
+        Create new  distributed  TF-IDF  transformer instance
+
+        Parameters
+        -----------
+
+        client : dask.distributed.Client optional Dask client to use
+        """
+        super(TfidfTransformer, self).__init__(
+            client=client, verbose=verbose, **kwargs
+        )
+
+        self.datatype = "cupy"
+
+        # Make any potential model args available and catch any potential
+        # ValueErrors before distributed training begins.
+        self._set_internal_model(s_TfidfTransformer(**kwargs))
+
+    @staticmethod
+    @with_cupy_rmm
```

Comment:
I had tested this at one point and recall decorators not getting propagated to the workers. My testing was a Dask versions ago. Have you tested this recently just to make sure? Just to be safe, we can use the context manager inside the function to guarantee it's used.

Reply:
So I just verified it and it seems to be working (see [gist](https://gist.github.com/VibhuJawa/451795f57489dba67a41c6f6700a7908)) 


But we can still skip it as we anyways decorate the underlying function too. 
https://github.com/rapidsai/cuml/pull/2698/files#diff-27c6ae1bf7c794b7e79b86dbddc0f6baR113-R131 . 

Happy to keep/remove it, let me know .

Reply:
Awesome! If it's working then I'm okay keeping it as it is.